boa_engine/src/error.rs

```diff
@@ -379,6 +379,13 @@ impl JsError {
         }
         self
     }
+
+    /// Is the [`JsError`] not catchable in JavaScript.
+    #[inline]
+    pub(crate) fn is_non_catchable(&self) -> bool {
```

Comment:
Would be better to implement it as `is_catchable`, I think. It's not common in Rust APIs to implement the negative of a property.